subt_ign/src/GameLogicPlugin.cc

```diff
@@ -640,6 +645,52 @@ void GameLogicPlugin::PostUpdate(
     std::map<std::string, ignition::math::Pose3d>::iterator baseIter =
       this->dataPtr->poses.find(subt::kBaseStationName);
 
+    _ecm.Each<gazebo::components::DetachableJoint>(
+        [&](const gazebo::Entity &,
+            const gazebo::components::DetachableJoint *_detach) -> bool
+        {
+          auto parentModel = _ecm.Component<gazebo::components::ParentEntity>(
+              _detach->Data().parentLink);
+          auto childModel = _ecm.Component<gazebo::components::ParentEntity>(
+              _detach->Data().childLink);
+
+          bool parentPerformer = false;
+          for (const auto &t :
+              _ecm.ChildrenByComponents(parentModel->Data(),
+                gazebo::components::Performer()))
+          {
+            parentPerformer = true;
+            break;
+          }
+
+          bool childPerformer = false;
+          for (const auto &t :
+              _ecm.ChildrenByComponents(childModel->Data(),
+                gazebo::components::Performer()))
+          {
+            childPerformer = true;
+            break;
+          }
+
+          auto parentName = _ecm.Component<gazebo::components::Name>(
```

Comment:
nit: `parentName` and `childName` are defined again below.

Reply:
Fixed.